storage/innobase/os/os0proc.cc

```diff
@@ -155,19 +188,39 @@ os_mem_alloc_large(
 	ut_ad(ut_is_2pow(size));
 	size = *n = ut_2pow_round(*n + (size - 1), size);
 	ptr = mmap(NULL, size, PROT_READ | PROT_WRITE,
-		   MAP_PRIVATE | OS_MAP_ANON, -1, 0);
-	if (UNIV_UNLIKELY(ptr == (void*) -1)) {
+		   MAP_PRIVATE | OS_MAP_ANON |
+		   (populate ? OS_MAP_POPULATE : 0), -1, 0);
+	if (UNIV_UNLIKELY(ptr == MAP_FAILED)) {
 		fprintf(stderr, "InnoDB: mmap(%lu bytes) failed;"
 			" errno %lu\n",
 			(ulong) size, (ulong) errno);
-		ptr = NULL;
+		return NULL;
 	} else {
 		os_fast_mutex_lock(&ut_list_mutex);
 		ut_total_allocated_memory += size;
 		os_fast_mutex_unlock(&ut_list_mutex);
 		UNIV_MEM_ALLOC(ptr, size);
 	}
 #endif
+
+#if OS_MAP_ANON && OS_MAP_POPULATE
+	/* MAP_POPULATE is only supported for private mappings
+	since Linux 2.6.23. */
+	populate = populate && !os_compare_release("2.6.23");
+
+	if (populate) {
+		fprintf(stderr, "InnoDB: Warning: mmap(MAP_POPULATE) "
```

Comment:
ib_errf in 5.6

Reply:
Based on its existing use, I should also create a new error code for this? (and then probably for 5.7 too)

Reply:
A new error code should not be required. Can you show an example of an existing use?

Reply:
ib_errf(m_trx->mysql_thd, IB_LOG_LEVEL_ERROR, ER_TABLE_SCHEMA_MISMATCH, ".ibd file has invlad table flags: %lx", ibd_table_flags);

So it needs a thread, a log level, an error code, and then a printf style message.

Reply:
Ah, my bad. I meant ib_logf(IB_LOG_LEVEL_WARN ... ), which essentially is the same fprintf(stderr)